providers/core.tcpsocket.js

```diff
@@ -75,6 +75,16 @@ Socket_firefox.prototype.write = function(buffer, continuation) {
   }
 };
 
+Socket_firefox.prototype.pause = function(continuation) {
+  this.clientSocket.pause();
```

Comment:
this would fail if called on a server socket, right? wrap in a catch or error early?